app/src/main/java/movierating/services/ReviewServiceImpl1.java

```diff
@@ -15,7 +18,21 @@ public ReviewServiceImpl1(UserRepository userRepository, MovieRepository movieRe
 
 
     @Override
-    public void addReview() {
+    public void addReview(String userName, String movieName, int rating) {
+
+        User user = this.userRepository.getUserList().get(userName);
+        if (user == null) {
```

Comment:
`
if (MapUtils.isNotEmpty(this.userRepository.getUsers()) && this.userRepository.getUsers().contains(userName)) {
 // do required
}
`
https://stackoverflow.com/questions/12721076/best-practice-to-validate-null-and-empty-collection-in-java

would be better choice.

Reply:
Dependency information
https://commons.apache.org/proper/commons-collections/dependency-info.html